components/ds18b20/src/ds18b20.c

```diff
@@ -134,7 +134,10 @@ esp_err_t ds18b20_get_temperature(ds18b20_device_handle_t ds18b20, float *ret_te
 
     const uint8_t lsb_mask[4] = {0x07, 0x03, 0x01, 0x00}; // mask bits not used in low resolution
     uint8_t lsb_masked = scratchpad.temp_lsb & (~lsb_mask[scratchpad.configuration >> 5]);
-    *ret_temperature = (((int16_t)scratchpad.temp_msb << 8) | lsb_masked)  / 16.0f;
+    // Combine the MSB and masked LSB into a signed 16-bit integer
+    int16_t temperature_raw = (((int16_t)scratchpad.temp_msb << 8) | lsb_masked);
+    // Convert the raw temperature to a float,
+    *ret_temperature = temperature_raw / 16.0f;
 
```

Comment:
I assume you tested this numericly; it appears that the crux of my fix is not implemented; using a `int8_t`?

```diff
-    *ret_temperature = (((int16_t)scratchpad.temp_msb << 8) | lsb_masked)  / 16.0f;
+    *ret_temperature = (((int8_t)scratchpad.temp_msb << 8) | lsb_masked)  / 16.0f;

```

Reply:
@morgana2313 Although you fix gets a correct result, we decided to implement it in canonical way.

The original  `(((int16_t)scratchpad.temp_msb << 8) | lsb_masked)` is implicitly converted to unsigned type in C. So we explicitly convert to `int16_t`:
```c
int16_t temperature_raw = (((int16_t)scratchpad.temp_msb << 8) | lsb_masked);
```

Reply:
Also, shift left an int8_t variable without expanding its data type is an undefined behavior in C.

Reply:
Very well!